opentelemetry-journald-logs/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+        self.message_size_limit = Some(message_size_limit);
+        self
+    }
+
+    pub fn attribute_prefix(mut self, attribute_prefix: Option<String>) -> Self {
+        if let Some(prefix) = attribute_prefix {
+            self.attribute_prefix = Some(prefix.to_uppercase());
+        } else {
+            self.attribute_prefix = None;
+        }
+        self
+    }
+
+    pub fn json_format(mut self, json_format: bool) -> Self {
+        self.json_format = json_format;
+        self
+    }
+
+    pub fn build(self) -> Result<JournaldLogExporter, &'static str> {
+        let identifier = self.identifier.ok_or("Identifier is required")?;
+        let message_size_limit = self
+            .message_size_limit
+            .ok_or("Message size limit is required")?;
+        Ok(JournaldLogExporter {
+            identifier: CString::new(identifier).map_err(|_| "Invalid identifier")?,
+            message_size_limit,
+            attribute_prefix: self.attribute_prefix,
+            json_format: self.json_format,
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct JournaldLogExporter {
+    identifier: CString,
+    message_size_limit: usize,
+    attribute_prefix: Option<String>,
+    json_format: bool,
+}
+
+impl JournaldLogExporter {
+    pub fn builder() -> JournaldLogExporterBuilder {
+        JournaldLogExporterBuilder::default()
+    }
+
+    fn send_to_journald(&self, iovecs: &[libc::iovec]) -> Result<(), std::io::Error> {
+        let ret = unsafe { sd_journal_sendv(iovecs.as_ptr(), iovecs.len() as c_int) };
+        if ret < 0 {
+            Err(std::io::Error::last_os_error())
+        } else {
+            Ok(())
+        }
+    }
+
+    fn send_log_to_journald(&self, log_data: &LogData) -> Result<(), std::io::Error> {
+        let mut iovecs: Vec<libc::iovec> = Vec::new();
+
+        // Add the SYSLOG_IDENTIFIER field
+        let identifier_str = format!("SYSLOG_IDENTIFIER={}", self.identifier.to_str().unwrap());
+        let identifier_field = CString::new(identifier_str).unwrap();
+        iovecs.push(libc::iovec {
+            iov_base: identifier_field.as_ptr() as *mut c_void,
+            iov_len: identifier_field.as_bytes().len(),
+        });
+
+        // Initialize potential message and attributes outside conditional blocks
+        let mut message: Option<CString> = None;
+        let mut attributes: Vec<CString> = Vec::new();
+
+        if self.json_format {
+            #[cfg(feature = "json")]
+            {
+                // Serialize message and attributes as JSON
+                let log_entry = LogEntry::from_log_data(log_data, self.attribute_prefix.clone());
+                let message_str = format!("MESSAGE={}", serde_json::to_string(&log_entry).unwrap());
+                message = Some(CString::new(message_str).unwrap());
+            }
+            #[cfg(not(feature = "json"))]
+            {
+                return Err(std::io::Error::new(
```

Comment:
Move this validation to `json_format` method where the user configures this bool variable?

Reply:
Done